packages/common/src/AncillaryDataParser.ts

```diff
@@ -0,0 +1,254 @@
+import Web3 from "web3";
+
+type CharObject = {
+  character: string;
+  escape: boolean;
+  skip: boolean;
+};
+
+/**
+ * @title Parse ancillary data.
+ * @notice Ancillary data parser implementation following guidelines at:
+ * https://docs.google.com/document/d/1zhKKjgY1BupBGPPrY_WOJvui0B6DMcd-xDR8-9-SPDw/edit
+ * @param {String} hex string representation of ancillaryData
+ * @return {Object} parsed ancillary data object.
+ */
+export function parseAncillaryData(ancillaryData: string): Record<string, unknown> {
+  // Some requesting contracts set the synthetic token address as ancillary data, so try to parse it first:
+  if (Web3.utils.isAddress(ancillaryData)) return { address: ancillaryData };
```

Comment:
is there any case that this will return the value when it is _not_ an address? this simply validates that: a) this is hex. b) the length is correct for an ethereum address c) the checksum on the case matches. is there a scenario where this can return for something that matches these criteria but is not an address?

Reply:
> is there a scenario where this can return for something that matches these criteria but is not an address?

I think this is an assumption we can make by using `Web3.utils.isAddress`

Reply:
Agreed with @nicholaspai that this assumption seems reasonable, and I'm not aware (per @Reinis-FRP's point) of any use case where `Web3.utils.isAddress` would not be adequate as a check or we could receive some ancillary data that passes the `isAddress` check but is not actually an address.